runtime/include/QuantumDevice.hpp

```diff
@@ -103,6 +104,20 @@ struct QuantumDevice {
      */
     [[nodiscard]] virtual auto GetDeviceShots() const -> size_t = 0;
 
+    /**
+     * @brief Set the PRNG of the device.
+     *
+     * @param gen The std::mt19937 PRNG object.
+     */
```

Comment:
I think we should document the expectations of the interface here:
```suggestion
    /**
     * @brief Set the PRNG of the device.
     *
     * The Catalyst runtime enables seeded program execution on non-hardware devices.
     * A random number generator instance is managed by the runtime to predictably
     * generate results for non-deterministic programs, such as those involving `Measure`
     * calls.
     * Devices implementing support for this feature do not need to use the provided
     * PRNG instance as their sole source of random numbers, but it is expected that the
     * the same instance state will predictable and reproducibly generate the same
     * program results. It is also expected that the provided PRNG state is evolved
     * sufficiently so that two device executions sharing the same instance do not produce
     * identical results.
     * The provided PRNG instance is not thread-locked, and devices wishing to share it
     * across threads will need to provide their own thread-safety.
     *
     * @param gen The std::mt19937 PRNG object.
     */
```

Reply:
good idea, done